include/dynd/types/tuple_type.hpp

```diff
@@ -145,13 +147,13 @@ namespace ndt {
     }
 
     /** Makes a tuple type with the specified types */
-    static type make(const nd::array &field_types, bool variadic = false)
+    static type make(std::vector<type> field_types, bool variadic = false)
     {
       return type(new tuple_type(field_types, variadic), false);
```

Comment:
Eventually, yes, but I wouldn't worry about these kind of optimizations for now. Let's get our code structure into a good place, then we can optimize that kind of thing.